tasmota/tasmota_xdrv_driver/xdrv_16_tuyamcu.ino

```diff
@@ -1197,157 +2035,240 @@ void TuyaInit(void) {
   int baudrate = 9600;
   if (Settings->flag4.tuyamcu_baudrate) { baudrate = 115200; }  // SetOption97 - Set Baud rate for TuyaMCU serial communication (0 = 9600 or 1 = 115200)
 
-  Tuya.buffer = (char*)(malloc(TUYA_BUFFER_SIZE));
-  if (Tuya.buffer != nullptr) {
-    TuyaSerial = new TasmotaSerial(Pin(GPIO_TUYA_RX), Pin(GPIO_TUYA_TX), 2);
-    if (TuyaSerial->begin(baudrate)) {
-      if (TuyaSerial->hardwareSerial()) { ClaimSerial(); }
-      // Get MCU Configuration
-      Tuya.SuspendTopic = true;
-      Tuya.ignore_topic_timeout = millis() + 1000; // suppress /STAT topic for 1000ms to avoid data overflow
-      AddLog(LOG_LEVEL_DEBUG, PSTR("TYA: Request MCU configuration at %d bps"), baudrate);
-
-      Tuya.heartbeat_timer = 0; // init heartbeat timer when dimmer init is done
-      return;
-    }
-    free(Tuya.buffer);
+  TuyaSerial = new TasmotaSerial(Pin(GPIO_TUYA_RX), Pin(GPIO_TUYA_TX), 2);
+  if (TuyaSerial->begin(baudrate)) {
+    if (TuyaSerial->hardwareSerial()) { ClaimSerial(); }
+    // Get MCU Configuration
+    pTuya->SuspendTopic = true;
+    pTuya->ignore_topic_timeout = millis() + 1000; // suppress /STAT topic for 1000ms to avoid data overflow
+    AddLog(LOG_LEVEL_DEBUG, PSTR("TYA: Request MCU configuration at %d bps"), baudrate);
+
+    pTuya->heartbeat_timer = 0; // init heartbeat timer when dimmer init is done
+    return;
   }
-  Tuya.active = false;
+  pTuya->active = false;
 }
 
-void TuyaSerialInput(void)
-{
-  while (TuyaSerial->available()) {
-    yield();
-    uint8_t serial_in_byte = TuyaSerial->read();
+// dump a buffer to debug
+void TuyaDump(unsigned char *buffer, int len){
+  char hex_char[(len * 2) + 2];
+  AddLog(LOG_LEVEL_DEBUG, 
+    PSTR("TYA: Raw Data: %s"), ToHex_P(buffer, len, hex_char, sizeof(hex_char)));
```

Comment:
Pls consider the change to %*_H instead of ToHex_P().

Suggested change: 
```
PSTR("TYA: Raw Data %*_H"), len, buffer);
```

Do this for all situations.